### DISCUSSION TITLE
filling apiv2 for harpy...

### DISCUSSION THREAD

**Discussion by idontveapc** [robertodoering/harpy#896] [General]
what should be filled for `callback uri / redirect url` and `website url`?

**Comment by robertodoering** [comment]
You can find a guide here: https://github.com/robertodoering/harpy/wiki/Twitter-api-key-setup

callback uri should be `harpy://` and website url doesn't matter

**Reply by idontveapc** [reply]
well, i deleted the app that gets created when you ```"apply-for-access"``` then we get to create ONLY ONE app.

so, i first created an app in v1.1 section and then in v2 added that app (created in v1.1).

and it let's me login (instead taking me to the website) but everything gets `403` errors. what is the reason?

**Reply by robertodoering** [reply]
I'm guessing you are trying to use the free api? You need to pay $100 a month to retrieve tweets using the api :^)
For the low price of $100 / month you will have access to retrieve a whopping 10,000!! tweets / month.

https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api

**Reply by idontveapc** [reply]
=(

**Reply by idontveapc** [reply]
they show how many request the free api made (0), so i thought they added free tier too...

also you get a *sign up free* button when **apply-for-access**

**Reply by idontveapc** [reply]
![Screenshot_20231031-134624](https://github.com/robertodoering/harpy/assets/122597295/8eecdf91-337f-4adb-ab72-4cda41f8d6ed)

**Reply by idontveapc** [reply]
![Screenshot_20231031-141531](https://github.com/robertodoering/harpy/assets/122597295/93757493-c2c9-4b86-a984-a4d663bb0707)

**Reply by idontveapc** [reply]
please reply show i can decide what to do

**Comment by idontveapc** [comment]
it takes me to Twitter's website, every time i tap authorize app

**Comment by ghost** [comment]
You have to pay for the Twitter API for it to provide data

**Reply by idontveapc** [reply]
you're saying the same thing.

**Reply by ghost** [reply]
I mean that it only works for posting not for viewing tweets on the free plan
